### DISCUSSION TITLE
Slow webhooks cause a halt in outgoing emails/queue workers

### DISCUSSION THREAD

**Discussion by olegbliaher** [postalserver/postal#2594] [Help with using Postal]
Hi there,

I've noticed one of my users has a webhook that reports all data to an endpoint, and it seems like the endpoint is very slow at taking in the POSTs from our Postal instance.

Some while ago I found over 2,000,000 rows in the `webhook_requests`-table, which caused all MySQL-queries to be really slow. I deleted all the rows that had a `created_at` older than some weeks, and that made the queue workers actually start [going through older emails that have been lying and waiting in the queue](https://github.com/postalserver/postal/discussions/2531).

Now, my question is - is there a way to separate queue workers from webhook workers?

**Comment by willpower232** [comment]
Unfortunately it is not possible to separate the tasks. You can increase the number of workers but at that volume, I'd suggest turning off the webhook until your user is able to resolve whatever issues they are struggling with.

If you're able to see any reasons for MySQL queries being slow within the webhook_requests table, I'm sure a PR would be appreciated :sweat_smile:
